### PR TITLE
Replace attempt library `go` with promise-utils

### PR DESCRIPTION
**Pull request by vponline**
This replaces the attempt library go-function implementation with api3/promise-utils.

### CODE REVIEW

File: packages/airnode-utilities/src/caching/index.ts [hunk 1]
```diff
@@ -65,18 +65,18 @@ const getKeys = (prefix?: string): string[] => {
...
+  const goRes = goSync(() => readFileSync(join(CACHE_BASE_PATH, key)));
+  if (!goRes.success) {
     logger.error(`Unable to read key data from fs cache`);
-    logger.error((e as Error).stack!);
+    logger.error(goRes.error.stack!);
+    return undefined;
   }
 
-  return undefined;
+  return JSON.parse(goRes.data.toString());
```
Comment (Siegrift) [inline packages/airnode-utilities/src/caching/index.ts]: JSON.parse can throw so this should be inside the `goSync` (similarly how it was previously in try-catch)

File: packages/airnode-utilities/src/caching/index.ts [hunk 2]
```diff
@@ -108,22 +107,33 @@ const addKey = (key: string, data: any, force = false) => {
...
-    const fileExists = existsSync(filePath);
-    if (fileExists && !force) {
-      logger.error(`Unable to overwrite key from fs cache as key exists and force is set to false`);
-      logger.error(`Key: ${key}`);
-      return;
-    }
 
-    if (fileExists) {
-      rmSync(filePath, { force: true });
+  const goFileExists = goSync(() => existsSync(filePath));
```
Comment (Siegrift) [inline packages/airnode-utilities/src/caching/index.ts:111]: Can this actually fail? I think `existsSync` does not throw an error.

File: packages/airnode-utilities/src/caching/index.ts [hunk 3]
```diff
@@ -133,51 +143,55 @@ const addKey = (key: string, data: any, force = false) => {
  * Sweep (garbage collect) the cache.
  */
 const sweep = () => {
-  try {
-    const now = Date.now();
+  const now = Date.now();
 
-    const stattedFiles = readdirSync(CACHE_BASE_PATH).map((file) => ({
+  const goRes = goSync(() =>
```
Comment (Siegrift) [inline packages/airnode-utilities/src/caching/index.ts:148]: This now changed the behaviour. Previously all of the code was inside the try-catch. 

EDIT: I see the rest of the code does not throw so it's probably fine.

File: packages/airnode-node/src/evm/utils.ts [hunk 4]
```diff
@@ -19,13 +20,17 @@ export function sortBigNumbers(bigNumbers: ethers.BigNumber[]) {
...
-  } catch {
-    return 'No revert string';
-  }
+
+  // Skip the funciton selector from the returned encoded data
+  // and only decode the revert reason string.
+  // Function selector is 4 bytes long and that is why we skip
+  // the first 2 bytes (0x) and the rest 8 bytes is the function selector
+  // return ethers.utils.defaultAbiCoder.decode(['string'], `0x${callData.substring(2 + 4 * 2)}`)[0];
+
```
Comment (Siegrift) [inline packages/airnode-node/src/evm/utils.ts:29]: Nit: remove `\n` as the comment is bound to that decode logic.

File: packages/airnode-deployer/src/infrastructure/index.ts [hunk 5]
```diff
@@ -207,7 +209,7 @@ async function terraformAirnodeManage(
 }
 
 // `maxConcurrency` field is required for deployment but missing for removal. This is the easiest way to type it.
-type CloudProviderExtended = CloudProvider & {
+export type CloudProviderExtended = CloudProvider & {
```
Comment (Siegrift) [inline packages/airnode-deployer/src/infrastructure/index.ts:212]: Do we need to export this?

**Comment by vponline** [comment]
This can be reviewed now. I managed to replace most of the try/catch blocks, but a few are left (mainly inside test files) and one [here](https://github.com/api3dao/airnode/blob/master/packages/airnode-examples/src/cli.ts#L11).

File: packages/airnode-deployer/src/infrastructure/index.ts [hunk 6]
```diff
@@ -45,24 +46,25 @@
 async function runCommand(command: string, options: CommandOptions) {
   const stringifiedOptions = JSON.stringify(options);
   const commandSpinner = logger.debugSpinner(`Running command '${command}' with options ${stringifiedOptions}`);
-  try {
-    const { stdout } = await exec(command, options);
-    commandSpinner.succeed(`Finished command '${command}' with options ${stringifiedOptions}`);
-    return stdout;
-  } catch (err) {
+
+  const goExec = await go(() => exec(command, options));
```
[the remark this answers is not in the document]
Comment (Siegrift) [reply packages/airnode-deployer/src/infrastructure/index.ts:50]: You can ignore this. I've spend some time investigating these alerts in the past and they were all false positive. I no longer consider these scanning alerts as that important.

File: packages/airnode-utilities/src/caching/index.ts (hunk 1 above)
Comment (Siegrift) [reply packages/airnode-utilities/src/caching/index.ts]: I see you reverted this in a separate commit. For the future, try to squash it with the previous commit.

Comment (vponline) [reply packages/airnode-utilities/src/caching/index.ts]: I'll keep this in mind :+1:

File: packages/airnode-deployer/src/infrastructure/index.ts (hunk 5 above)
Comment (vponline) [reply packages/airnode-deployer/src/infrastructure/index.ts:212]: I was getting a type error without including it [here](https://github.com/api3dao/airnode/pull/1276/files#diff-13b7c9bcdda766f79084f5c09740a68e14cda9410e225e5d9bc44d643f332562R80). It's due to the nodeSettings.cloudProvider having the option of 'local', but that is not allowed once deploying.

File: packages/airnode-utilities/src/caching/index.ts (hunk 2 above)
Comment (vponline) [reply packages/airnode-utilities/src/caching/index.ts:111]: I think you are right. I'll remove this.

File: packages/airnode-examples/test/example-files-are-up-to-date.test.ts [hunk 7]
```diff
@@ -11,15 +12,12 @@ describe('Verifies that all config.example.json files are up to date', () => {
...
-        const createConfig = await import(join(folder, 'create-config.ts'));
-        await createConfig.default(true);
-      } catch (e) {
-        throw new Error((e as Error).message);
-      }
+      const goCreateConfig = await go(() =>
+        import(join(folder, 'create-config.ts')).then((createConfig) => createConfig.default(true))
+      );
+
+      if (!goCreateConfig.success) throw goCreateConfig.error.message;
```
Comment (Siegrift) [inline packages/airnode-examples/test/example-files-are-up-to-date.test.ts:19]: We should be throwing only Errors not strings.

File: packages/airnode-adapter/e2e/index.ts [hunk 8]
```diff
@@ -294,17 +291,16 @@ describe('Extraction, encoding and simple on chain decoding', () => {
...
             _path: 'big.tooLargeDecimal',
           })
-        );
-        expect.fail();
-      } catch (e: any) {
-        expect(e.message).to.contain('out-of-bounds');
-      }
+        )
+      );
+      expect(goExtractAndEncode.success).to.equal(false);
```
Comment (Siegrift) [inline packages/airnode-adapter/e2e/index.ts:302]: You can use `assertGoError` instead of this assertion and the typecast below. (same for others).

**Review by Siegrift — COMMENTED** [review]
Minor comments, but otherwise LGTM

File: packages/airnode-adapter/e2e/index.ts (hunk 8 above)
Comment (vponline) [reply packages/airnode-adapter/e2e/index.ts]: This is cool, thanks for pointing it out!

**Review by Siegrift — APPROVED** [review]
👍 LGTM

**Pull request merged**